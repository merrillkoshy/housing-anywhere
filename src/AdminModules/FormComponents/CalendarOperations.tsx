import React, { useState } from "react";
import { Form, Dropdown } from "react-bootstrap";

const CalendarOperations = () => {
	const [type, setType] = useState(0);
	const [asType, selectType] = useState("");
	const [dateFrom, setDateFrom] = useState("");
	const [dateTo, setDateTo] = useState("");
	return (
		<>
			<Form.Group className="mb-3">
				<Form.Label>Type</Form.Label>
				<Dropdown>
					<Dropdown.Toggle variant="primary" id="type">
						{asType}
					</Dropdown.Toggle>

					<Dropdown.Menu>
						<Dropdown.Item
							onClick={(e) => {
								e.preventDefault();
								setType(0);
								selectType("MarkAvailable");
							}}
						>
							Available
						</Dropdown.Item>
						<Dropdown.Item
							onClick={(e) => {
								e.preventDefault();
								setType(1);
								selectType("MarkUnavailable");
							}}
						>
							Unavailable
						</Dropdown.Item>
					</Dropdown.Menu>
				</Dropdown>
			</Form.Group>
			<Form.Group className="mb-3">
				<Form.Label>Date From</Form.Label>
				<Form.Control
					type="text"
					id="dateFrom"
					value={dateFrom}
					pattern={"/^d{4}-d{2}-d{2}/"}
					onChange={(e) => {
						e.preventDefault();
						setDateFrom(e.currentTarget.value);
					}}
					placeholder="YYYY-MM-DD"
				/>
			</Form.Group>
			<Form.Group className="mb-3">
				<Form.Label>Date To</Form.Label>
				<Form.Control
					type="text"
					id="dateTo"
					value={dateTo}
					pattern={"/^d{4}-d{2}-d{2}/"}
					onChange={(e) => {
						e.preventDefault();
						setDateTo(e.currentTarget.value);
					}}
					placeholder="YYYY-MM-DD"
				/>
			</Form.Group>
		</>
	);
};
export default CalendarOperations;
