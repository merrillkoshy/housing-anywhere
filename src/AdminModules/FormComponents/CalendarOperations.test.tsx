import React from "react";
import { shallow } from "enzyme";
import CalendarOperations from "./CalendarOperations";

describe("CostsModule module", () => {
	const CalendarOperationsModule = shallow(<CalendarOperations />);
	it("renders", () => {
		expect(CalendarOperationsModule).toMatchSnapshot();
	});

	it("finds a type input which is a number", () => {
		expect(CalendarOperationsModule.find("#type").exists()).toBe(true);
	});
	it("finds a dateFrom input which is a string", () => {
		expect(CalendarOperationsModule.find("#dateFrom").exists()).toBe(true);
	});
	it("finds a dateTo input which is a string", () => {
		expect(CalendarOperationsModule.find("#dateTo").exists()).toBe(true);
	});
});
