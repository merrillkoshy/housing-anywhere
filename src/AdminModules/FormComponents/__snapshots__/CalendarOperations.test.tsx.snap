// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CostsModule module renders 1`] = `
<Fragment>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Type
    </FormLabel>
    <Dropdown
      navbar={false}
    >
      <DropdownToggle
        id="type"
        variant="primary"
      />
      <DropdownMenu
        align="left"
        alignRight={false}
        flip={true}
      >
        <DropdownItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          disabled={false}
          onClick={[Function]}
        >
          Available
        </DropdownItem>
        <DropdownItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          disabled={false}
          onClick={[Function]}
        >
          Unavailable
        </DropdownItem>
      </DropdownMenu>
    </Dropdown>
  </FormGroup>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Date From
    </FormLabel>
    <FormControl
      id="dateFrom"
      onChange={[Function]}
      pattern="/^d{4}-d{2}-d{2}/"
      placeholder="YYYY-MM-DD"
      type="text"
      value=""
    />
  </FormGroup>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Date To
    </FormLabel>
    <FormControl
      id="dateTo"
      onChange={[Function]}
      pattern="/^d{4}-d{2}-d{2}/"
      placeholder="YYYY-MM-DD"
      type="text"
      value=""
    />
  </FormGroup>
</Fragment>
`;
